secrets.go

```diff
@@ -0,0 +1,182 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/buildkite/go-pipeline/ordered"
+	"gopkg.in/yaml.v3"
+)
+
+var _ interface {
+	json.Unmarshaler
+	ordered.Unmarshaler
+	yaml.Marshaler
+} = (*Secrets)(nil)
+
+// Secrets is a sequence of secrets. It is useful for unmarshaling.
+type Secrets []Secret
+
+// UnmarshalOrdered unmarshals Secrets from []any (sequence of secret names).
+func (s *Secrets) UnmarshalOrdered(o any) error {
+	switch o := o.(type) {
+	case nil:
+		// `secrets: null` is invalid - should be omitted entirely or use valid formats
+		return fmt.Errorf("unmarshaling secrets: secrets cannot be null")
+
+	case *ordered.Map[string, interface{}]:
+		// Handle map syntax: {"ENV_VAR": "SECRET_KEY"}
+		return o.Range(func(envVar string, secretKeyVal interface{}) error {
```

Comment:
The short form of `interface{}` is `any`, which is now pretty standard:

```suggestion
	case *ordered.Map[string, any]:
		// Handle map syntax: {"ENV_VAR": "SECRET_KEY"}
		return o.Range(func(envVar string, secretKeyVal any) error {
```